File: api-tester/src/main/java/com/revenuecat/apitester/java/TestStoreProductAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.models.Period;
import com.revenuecat.purchases.models.Price;
import com.revenuecat.purchases.models.StoreProduct;
import com.revenuecat.purchases.models.TestStoreProduct;

@SuppressWarnings({"unused", "deprecation"})
final class TestStoreProductAPI {
    static void checkConstructors(final Price price, final Period period) {
        new TestStoreProduct(
                "id", "title", "description", price, period, null, null
        );
        new TestStoreProduct(
                "id",  "title", "description", price, period, period, price
        );
        new TestStoreProduct(
             "id", "name", "title", "description", price, period, null, null
        );
        new TestStoreProduct(
            "id", "name", "title", "description", price, period, period, price
        );
    }

    static void checkTestStoreProductIsStoreProduct(final TestStoreProduct testStoreProduct) {
        StoreProduct storeProduct = testStoreProduct;
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/TestStoreProductAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.models.Period
import com.revenuecat.purchases.models.Price
import com.revenuecat.purchases.models.StoreProduct
import com.revenuecat.purchases.models.TestStoreProduct

@Suppress("unused", "UNUSED_VARIABLE", "LongMethod", "DEPRECATION")
private class TestStoreProductAPI {
    fun checkConstructor(price: Price, period: Period?) {
        TestStoreProduct(
            id = "ID",
            title = "title",
            price = price,
            description = "description",
            period = period,
        )
        TestStoreProduct(
            id = "ID",
            title = "title",
            price = price,
            description = "description",
            period = period,
            freeTrialPeriod = period,
        )
        TestStoreProduct(
            id = "ID",
            title = "title",
            price = price,
            description = "description",
            period = period,
            introPrice = null,
        )
        TestStoreProduct(
            id = "ID",
            title = "title",
            price = price,
            description = "description",
            period = period,
            introPrice = price,
        )
        TestStoreProduct(
            id = "ID",
            name = "name",
            title = "title",
            price = price,
            description = "description",
            period = period,
        )
        TestStoreProduct(
            id = "ID",
            name = "name",
            title = "title",
            price = price,
            description = "description",
            period = period,
            freeTrialPeriod = period,
        )
        TestStoreProduct(
            id = "ID",
            name = "name",
            title = "title",
            price = price,
            description = "description",
            period = period,
            introPrice = null,
        )
        TestStoreProduct(
            id = "ID",
            name = "name",
            title = "title",
            price = price,
            description = "description",
            period = period,
            introPrice = price,
        )
    }

    fun checkTestStoreProductIsStoreProduct(testStoreProduct: TestStoreProduct) {
        val storeProduct: StoreProduct = testStoreProduct
    }
}
